[nnc] Tests for proposed feature: loop bounds conditional simplification

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54121 [nnc] Tests for proposed feature: loop bounds conditional simplification**
* #54094 [nnc] Test for using int64 dimensions

It would be nice to do range analysis to determine if a condition
cannot be satisfied.  These are some tests that we should be able to turn on
once we have this feature.

Differential Revision: [D27107956](https://our.internmc.facebook.com/intern/diff/D27107956/)

bertmaher committed:
[nnc] Tests for proposed feature: loop bounds conditional simplification

It would be nice to do range analysis to determine if a condition
cannot be satisfied.  These are some tests that we should be able to turn on
once we have this feature.

Differential Revision: [D27107956](https://our.internmc.facebook.com/intern/diff/D27107956/)

[ghstack-poisoned]

ZolotukhinM commented on test/cpp/tensorexpr/test_simplify.cpp:
```diff
@@ -4633,5 +4633,97 @@ TEST(Simplify, SimplifyBroadcastTermExpander) {
+  int N = 8;
+  Placeholder b("b", kFloat, {N});
+  VarHandle n("n", kInt);
+  Stmt* s =
+      For::make(n, 1, N, b.store({n}, IfThenElse::make(n < 1, 0.f, 1.0f)));
```
Nit: adding a comment showing the original IR would be helpful.

ZolotukhinM reviewed (approved):
👍
![image](https://user-images.githubusercontent.com/4588111/111411893-9fdace00-8698-11eb-93e1-c4a4fd496014.png)

navahgar commented:
Added a new issue for this: https://github.com/pytorch/pytorch/issues/54123

bertmaher committed:
Update on "[nnc] Tests for proposed feature: loop bounds conditional simplification"


It would be nice to do range analysis to determine if a condition
cannot be satisfied.  These are some tests that we should be able to turn on
once we have this feature.

Differential Revision: [D27107956](https://our.internmc.facebook.com/intern/diff/D27107956/)

[ghstack-poisoned]

bertmaher committed:
fix lint on "[nnc] Tests for proposed feature: loop bounds conditional simplification"


It would be nice to do range analysis to determine if a condition
cannot be satisfied.  These are some tests that we should be able to turn on
once we have this feature.

Differential Revision: [D27107956](https://our.internmc.facebook.com/intern/diff/D27107956/)

[ghstack-poisoned]
